src/Algolia.Search.Test/EndToEnd/Recommendation/RecommendationClientTest.cs

```diff
@@ -58,7 +58,11 @@ public void TestRecommendationClient()
             {
                 BaseTest.RecommendationClient.SetPersonalizationStrategy(request);
             }
-            catch (AlgoliaApiException e) when (e.HttpErrorCode != 429)
+            catch (AlgoliaApiException e) when ((e.HttpErrorCode == 429) && (e.Message == "{\"status\":429,\"message\":\"Number of strategy saves exceeded for the day\"}"))
```

Comment:
Please remove the test on the message and keep the test on the code. The reasoning behind this is that the code won't message but the message is likely to evolve as the API doesn't commit on it.